docs/setup/settings.asciidoc

```diff
@@ -26,12 +26,15 @@ which may cause a delay before pages start being served.
 Set to `false` to disable Console. *Default: `true`*
 
 | `csp.rules:`
- | deprecated:[7.14.0,"In 8.0 and later, this setting will no longer be supported."]
+| deprecated:[7.14.0,"In 8.0 and later, this setting will no longer be supported."]
 A https://w3c.github.io/webappsec-csp/[Content Security Policy] template
 that disables certain unnecessary and potentially insecure capabilities in
 the browser. It is strongly recommended that you keep the default CSP rules
 that ship with {kib}.
 
+| `csp.unsafe_eval`
+| Set this to `false` to remove the https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src#unsafe_eval_expressions[`unsafe-eval`] source expression from the `script-src` directive. *Default: `true`*
```

Comment:
I think it would be beneficial to describe the impact of turning this on (e.g., risk of  Handlebars incompatibility, and a small change that other features may not work as expected).

What do you (all) think about marking this as an experimental feature, initially? It would provide us some cover in the event that we discover an unknown incompatibility in the wild.

Reply:
> I think it would be beneficial to describe the impact of turning this on (e.g., risk of Handlebars incompatibility, and a small change that other features may not work as expected).

Yeah I was considering it, but was afraid that I couldn't find a good way to phrase it so it made sense. I'll give it a shot!

> What do you (all) think about marking this as an experimental feature, initially?

Good idea. It might slow the adoption, which in turn means that we might not discover come edge cases. But the benefits of marking it as experimental outweighs that one downside IMO. Will we be able to set it to `false` for new clusters in cloud by default if it's experimental?

Reply:
> I think it would be beneficial to describe the impact of turning this on (e.g., risk of Handlebars incompatibility, and a small change that other features may not work as expected).

+1

> What do you (all) think about marking this as an experimental feature, initially? It would provide us some cover in the event that we discover an unknown incompatibility in the wild.

+1

Reply:
I've updated both the release notes in the description above and the settings docs (in this commit: https://github.com/elastic/kibana/pull/124484/commits/c2a067baf90918270247b47f335d580c839d0e93) to both mention that this is experimental and to let the user know the impact of tuning this feature off.